File: double/DoubleTheArrayElementsFunctional.java
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class DoubleTheArrayElementsFunctional {
    public static void main(String[] args) {
        List<Integer> numbers = Arrays.asList(1, 2, 3, 4, 5);

        List<Integer> doubledNumbers = numbers.stream()
                .map(number -> number * 2)
                .collect(Collectors.toList());

        System.out.println(doubledNumbers);
    }
}

File: double/DoubleTheArrayElementsFunctional.ts
function doubleTheArrayElementsFunctional(numbers: number[]): number[] {
  return numbers.map((number) => number * 2);
}

const numbers = [1, 2, 3, 4, 5];
const doubledNumbers = doubleTheArrayElementsFunctional(numbers);
console.log(doubledNumbers);
